libs/api/domains/occupational-licenses/src/lib/models/occupationalLicense.model.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  ObjectType,
+  Field,
+  ID,
+  InterfaceType,
+  registerEnumType,
+} from '@nestjs/graphql'
+
+export enum OccupationalLicenseType {
+  EDUCATION = 'EDUCATIONAL',
+  HEALTH = 'HEALTH_DIRECTORATE',
+}
+
+registerEnumType(OccupationalLicenseType, {
+  name: 'OccupationalLicenseType',
+})
+
+@InterfaceType({
+  resolveType(license: OccupationalLicense) {
+    if (license.institution === 'HEALTH_DIRECTORATE') {
+      return HealthDirectorateLicense
+    }
+    return EducationalLicense
+  },
+})
+export abstract class OccupationalLicense {
+  @Field((type) => OccupationalLicenseType)
+  institution!: OccupationalLicenseType
+
+  @Field(() => ID)
+  id!: string
+  @Field(() => String)
+  type!: string
+
+  @Field(() => String)
+  profession!: string
+
+  @Field(() => Boolean)
+  isValid!: boolean
+
+  @Field(() => String)
+  validFrom!: string
+}
+
+@ObjectType('OccupationalLicensesEducationalLicense', {
+  implements: OccupationalLicense,
+})
+export class EducationalLicense extends OccupationalLicense {
+  @Field(() => String)
+  downloadUrl?: string
```

Comment:
Yes I had it optional because its only generated when the user goes to detail view of the license, its not available in the overview call or so to say.